include/CLI/Validators.hpp

```diff
@@ -191,6 +191,33 @@ struct Range : public Validator {
     template <typename T> explicit Range(T max) : Range(static_cast<T>(0), max) {}
 };
 
+namespace detail {
+/// split a string into a program name and command line arguments
+/// the string is assumed to contain a file name followed by other arguments
+/// the return value contains is a pair with the first argument containing the program name and the second everything
+/// else
+inline std::pair<std::string, std::string> split_program_name(std::string commandline) {
```

Comment:
Why is this is Validators.hpp? Shouldn't it be in one of the other headers, like string tools?

Reply:
The main reason was it uses the validators to check for an existing file, and if it was included in one of the other tools that header would then have to include the validators header,  which I wasn't sure how you felt about.

Reply:
Okay, we'll leave it there for now. Maybe should be moved to a new header at some point.